go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -157,7 +157,12 @@ def run_go_get(dependencies = [])
           tmp_go_file = "#{SecureRandom.hex}.go"
 
           package = Dir.glob("[^\._]*.go").any? do |path|
-            !File.read(path).include?("// +build")
+            # The build syntax changed from 1.16-1.18:
+            # https://go.dev/design/draft-gobuild
+            # https://go.dev/doc/go1.18#go-command
+            # old syntax: // +build
+            # new syntax: //go:build
+            !File.read(path).include?("/(\/\/ \+build)|(\/\/go:build)/".freeze)
```

Comment:
I found this regex a bit inscrutable, so open to suggestions... 

I could pull this out into an array of regexes like most other string matching, or maybe Ruby has something as simple as "include?(x).or(y)"??

Reply:
What about this?

```ruby
!File.read(path).match?(%r((// \+build)|(//go:build)))
```

Reply:
Could also pull out the regular expression into a constant with a name but 🤷